### PR TITLE
Allow to select user scripts in the Add Script modal

### LINKED ISSUE #1918: Select scripts from dropdown
See https://discuss.webknossos.org/t/scripts-quick-select/350/4

I would expand the existing "Add script" modal to include a dropdown for selecting a script. Once an item is selected from the dropdown, the textbox gets filled with the code and the user can apply the script as before.

<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/1918/create?referer=github" target="_blank">Log Time</a>

### PR DESCRIPTION
**Pull request by daniel-wer**
See title.
I would love to add syntax highlighting to that textarea, but haven't found a simple solution. Not sure whether it's used that much and we should include another clunky lib to accomplish this.

### Mailable description of changes:
 - Added the possibility to select a pre-defined user script in the "Add scripts" modal

### Steps to test:
- Add a user script in the administration interface. Check whether you can select it in the Add scripts modal.
- Modify the script in the modal and try to select another script - a confirm dialog should appear
- Check whether tasks with attached scripts still execute the script successfully

### Issues:
- fixes #1918

------
- [x] Ready for review

### CODE REVIEW

File: app/assets/javascripts/oxalis/view/user_scripts_modal.js [hunk 1]
```diff
@@ -43,7 +78,26 @@ class UserScriptsModalView extends React.PureComponent {
         onOk={this.handleClick}
         onCancel={this.props.onClose}
       >
-        <Input type="textarea" rows={4} onChange={this.handleCodeChange} value={this.state.code} />
+        <Spin spinning={this.state.isLoading}>
+          <Select
+            value={this.state.selectedScript}
+            style={{ width: 200, "margin-bottom": 10 }}
```
Comment (hotzenklotz) [inline app/assets/javascripts/oxalis/view/user_scripts_modal.js]: You probably meant `marginBottom`

File: app/assets/javascripts/libs/gist.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/*
+ * gist.js
+ * @flow
+ */
+
+import _ from "lodash";
```
Comment (hotzenklotz) [inline app/assets/javascripts/libs/gist.js]: Good idea to move that into a separate file :-)

File: app/assets/javascripts/oxalis/view/user_scripts_modal.js [hunk 3]
```diff
@@ -14,14 +16,47 @@ class UserScriptsModalView extends React.PureComponent {
...
+      isCodeChanged: true,
     });
   };
 
+  handleScriptChange = async (scriptId: string) => {
+    const script = this.state.scripts.find(s => s.id === scriptId);
+    if (script == null) return;
+    if (
+      !this.state.isCodeChanged ||
+      confirm("This will replace the code you've written. Continue?")
```
Comment (hotzenklotz) [inline app/assets/javascripts/oxalis/view/user_scripts_modal.js]: Please move the text to `message.js` file. Also, please spell out `have`.

I would love to see antd's  `Modal.Confirm` here. https://ant.design/components/modal/

File: app/assets/javascripts/oxalis/view/user_scripts_modal.js (hunk 1 above)
Comment (daniel-wer) [inline app/assets/javascripts/oxalis/view/user_scripts_modal.js]: Not sure what the difference is, works either way.

Comment (hotzenklotz) [inline app/assets/javascripts/oxalis/view/user_scripts_modal.js]: You get a big fat warning for React this way. React expects camelCased CSS keywords.

**Comment by hotzenklotz** [comment]
Read only is not a good idea That would prevent copy paste for scripts. I agree not many people will code in this modal, but we should not prevent that. That would counter intuitive with the "Add Script" modal. 

@daniel-wer BTW, while we are at it. Do you mind doing one more thing. Please remove the "Add Script" link from the global menu bar and move into the dropdown actions next to the "Save" button. Thank you.

**Comment by daniel-wer** [comment]
Hmm I'm not sure about making the input box read-only as originally we introduced this modal so users could add their scripts without opening the dev console and making it read-only would defeat that purpose. But we could ask the lab. Maybe this would make it easier for them to make sure that everyone uses the same script (the one officially added by a lab admin).

Changing the dropdown value to "Modified script" or "Unsaved script" would be fine with me.

@hotzenklotz Yes I'll move it there.

**Comment by daniel-wer** [comment]
Thanks for your helpful feedback, I applied all of it. If you give the thumbs up, I'll merge :)

**Pull request merged**